api/api.go

```diff
@@ -2,10 +2,65 @@ package api
 
 import (
 	"context"
+	"fmt"
 )
 
-// LogFunc logs a message to the host's logs.
-type LogFunc func(context.Context, string)
+// LogLevel controls the volume of logging. The lower the number the more
+// detail is logged.
+//
+// Note: The most voluminous level, LogLevelDebug is -1 to prevent users from
+// accidentally defaulting to it.
+type LogLevel int32
+
+const (
+	LogLevelDebug LogLevel = iota - 1
```

Comment:
Do you think it helps reinforce that the ordinals must not change by `=` them all instead of `iota`? Notably if Trace was added later for some reason. Otherwise a code comment could be good

Reply:
= them all sounds good. I can do that also for the other things